flockoff/validator/trainer.py

```
@@ -39,6 +39,30 @@ def download_dataset(
+                    f"Please use a specific commit hash for the revision. "
+                    f"To disable this check, set the environment variable VALIDATOR_CHECK_COMMIT_HASH=false."
+                )
+            bt.logging.info(f"Revision '{revision}' is a valid commit hash.")
+        except Exception as e:
+            # Catch HfHubHTTPError if revision doesn't exist or other API errors
+            bt.logging.error(f"Error fetching repo info for {namespace}@{revision}: {e}")
+            raise ValueError(
+                f"Failed to validate revision '{revision}' for dataset '{namespace}'. "
+                f"Ensure the revision exists and is accessible. Original error: {e}. "
+                f"To disable this check, set the environment variable VALIDATOR_CHECK_COMMIT_HASH=false."
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)